### PR TITLE
Introducing Discounting & Warmup Phase

### PR DESCRIPTION
**Pull request by oliver-scheel**

### CODE REVIEW

File: l5kit/l5kit/planning/vectorized/closed_loop_model.py [hunk 1]
```diff
@@ -254,8 +274,17 @@ def forward(self, data_batch: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor
                 raise NotImplementedError("Loss function is undefined.")
 
             target_weights = data_batch["target_availabilities"][:, :future_num_frames]
+            # Only calculate loss for the correct frames, i.e. not during the warmup phase,
+            # and only up to self.unroll_num_frames (in case more frames are loaded).
+            target_weights[:, :start_idx_to_consider] = 0
+            target_weights[:, end_idx_to_consider:] = 0
             target_weights = target_weights.unsqueeze(-1) * self.weights_scaling
 
+            if self.discount_factor < 1:
```
Comment (lucabergamini) [inline l5kit/l5kit/planning/vectorized/closed_loop_model.py:283]: this should work also for 1 right?

File: l5kit/l5kit/planning/vectorized/closed_loop_model.py [hunk 2]
```diff
@@ -143,6 +156,13 @@ def forward(self, data_batch: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor
         yaw_t0_from_ts = zero
         yaw_ts_from_t0 = zero
 
+        # With self.warmup_num_frames > 0 we introduce an initial "sampling" period, meaning
+        # we simulate sampling from the model distribution by following it for a random
+        # number of steps in [0, self.warmup_num_frames) -> calculate corresponding start /
+        # end indices here for unrolling / loss calculation.
+        start_idx_to_consider = int(torch.randint(0, self.warmup_num_frames + 1, (1,)).item())
+        end_idx_to_consider = start_idx_to_consider + self.unroll_num_frames
```
Comment (lucabergamini) [inline l5kit/l5kit/planning/vectorized/closed_loop_model.py:164]: let's make this fixed and remove the additional param

File: l5kit/l5kit/tests/artefacts/config_vectorized.yaml [hunk 3]
```diff
@@ -10,7 +10,10 @@ model_params:
   disable_other_agents: False
   disable_map: False
   disable_lane_boundaries: True
-  gobal_head_dropout: 0.0
+  global_head_dropout: 0.0
```
Comment (lucabergamini) [inline l5kit/l5kit/tests/artefacts/config_vectorized.yaml:13]: thanks for catching this!

**Review by lucabergamini — APPROVED** [review]
nits but LGTM

File: l5kit/l5kit/planning/vectorized/closed_loop_model.py (hunk 1 above)
Comment (oliver-scheel) [reply l5kit/l5kit/planning/vectorized/closed_loop_model.py:283]: Yeah, removing the if

**Pull request merged**